lib/identifiers/doi.rb

```diff
@@ -2,42 +2,31 @@ module Identifiers
   class DOI
     REGEXP = %r{
       \b
-      10 # Directory indicator (always 10)
+      10                                      # Directory indicator (always 10)
       \.
       (?:
         # ISBN-A
-        97[89]\. # ISBN (GS1) Bookland prefix
-        \d{2,8}  # ISBN registration group element and publisher prefix
-        /        # Prefix/suffix divider
-        \d{1,7}  # ISBN title enumerator and check digit
+        97[89]\.                              # ISBN (GS1) Bookland prefix
+        \d{2,8}                               # ISBN registration group element and publisher prefix
+        /                                     # Prefix/suffix divider
+        \d{1,7}                               # ISBN title enumerator and check digit
         |
         # DOI
-        \d{4,9}       # Registrant code
-        /             # Prefix/suffix divider
-        [^[:space:]]+ # DOI suffix
+        \d{4,9}                               # Registrant code
+        /                                     # Prefix/suffix divider
+        (?:
+          # DOI suffix
+          [^[:space:]]+2-\#                   # Early Wiley suffix
+          |
+          [^[:space:]]+\([^[:space:]]+\)      # Suffix ending in balanced parentheses
```

Comment:
Actually - I found another problem. This will actually also match `10.1130/2013.2502(04))`. I think here, what we want is: `[^[:space:]]+\([^[:space:])]+\)`